cinema/views.py

```diff
@@ -101,8 +103,20 @@ def get_serializer_class(self):
         if self.action == "retrieve":
             return MovieDetailSerializer
 
+        if self.action == "upload_image":
+            return MovieImageSerializer
+
         return MovieSerializer
 
+    @action(detail=True, methods=["POST"], url_path="upload-image")
+    def upload_image(self, request, pk=None):
+        item = self.get_object()
+        serializer = self.get_serializer(item, data=request.data)
+
+        if serializer.is_valid(raise_exception=True):
```

Comment:
You don't need to use if-statement when you pass raise_exception=True to is_valid. It will raise exception automatically